modyn/trainer_server/internal/trainer/pytorch_trainer.py

```diff
@@ -244,7 +258,9 @@ def train(self) -> None:  # pylint: disable=too-many-locals, too-many-branches
         self._info("Handled OnBegin Callbacks.")
 
         batch_number = 0
-        for _ in range(self.epochs_per_trigger):
+        for epoch in range(self.epochs_per_trigger):
+            if self._downsampling_enabled and self._sample_before_batch and epoch % self._downsampling_period == 0:
```

Comment:
what if _downsampling_period == 0? (or how can I configure Modyn to only do downsampling once per trigger training set?)